yarp/yarp_compiler.c

```diff
@@ -516,6 +516,58 @@ yp_compile_class_path(LINK_ANCHOR *const ret, rb_iseq_t *iseq, const yp_node_t *
     }
 }
 
+/**
+ * In order to properly compile multiple-assignment, some preprocessing needs to
+ * be performed in the case of call or constant path targets. This is when they
+ * are read, the "parent" of each of these nodes should only be read once (the
+ * receiver in the case of a call, the parent constant in the case of a constant
+ * path).
+ */
+static void
+yp_compile_multi_write_lhs(rb_iseq_t *iseq, int lineno, const yp_node_t *node, LINK_ANCHOR *const ret, yp_compile_context_t *compile_context, uint8_t *pushed, bool nested) {
+    NODE dummy_line_node = generate_dummy_line_node(lineno, lineno);
```

Comment:
Can we pass in the `dummy_line_node` itself as an arg to `yp_compile_multi_write_lhs` avoid continuing to create it on recursion since we're not using the `lineno` isolated anywhere?